### ISSUE TITLE
[Web] Text rendering incorrectly when inside a Row with an ImageIcon

### ISSUE THREAD

**Issue by esDotDev** [flutter/flutter#56971]
We are seeing an issue in the latest master: v1.19.0-1

When a Image with an assigned color is in a row with some text, the Text is vertically displaced outside the row:
![image](https://user-images.githubusercontent.com/736973/81639039-012a7480-93d8-11ea-8380-fca8c4ae3d31.png)

This code reproduces it:
```
Row(children: [Image.asset("assets/icons/icon-edit.png", color: Colors.black), Text("FOO")])
```
v1.19.0-0: 
![image](https://user-images.githubusercontent.com/736973/81640063-d988db80-93da-11ea-991e-4acee16074c6.png)

v1.19.0-1: 
![image](https://user-images.githubusercontent.com/736973/81640089-e73e6100-93da-11ea-8832-6fb8df392960.png)

Notes:
* Only occurs when Image is in Row, and has a color assigned
* Only affects Text, other children in the row render fine

**Comment by TahaTesser** [comment]
Issue reproduces on latest `dev` and `master` channel but not on `beta` channel

#### `dev` channel
<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.19.0-0.0.pre, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.19.0-0.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision a849daf283 (4 days ago), 2020-05-07 18:59:02 -0700
    • Engine revision 3953c3ccd1
    • Dart version 2.9.0 (build 2.9.0-5.0.dev 4da5b40fb6)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (5 available)
    • Android SDK built for x86 • emulator-5554             • android-x86    •
      Android 10 (API 29) (emulator)
    • Taha’s iPhone             • 00008020-001059882212002E • ios            •
      iOS 13.4.1
    • macOS                     • macOS                     • darwin-x64     •
      Mac OS X 10.15.4 19E287
    • Web Server                • web-server                • web-javascript •
      Flutter Tools
    • Chrome                    • chrome                    • web-javascript •
      Google Chrome 81.0.4044.138

• No issues found!
``` 

</details> 


#### `master` channel
<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel master, 1.19.0-1.0.pre.43, on Mac OS X 10.15.4 19E287,
    locale en-GB)
    • Flutter version 1.19.0-1.0.pre.43 at /Users/tahatesser/Code/flutter_master
    • Framework revision c969b8af7b (15 hours ago), 2020-05-11 15:28:10 -0700
    • Engine revision ada8a0fd64
    • Dart version 2.9.0 (build 2.9.0-8.0.dev a6d06b59ec)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (4 available)
    • Taha’s iPhone • 00008020-001059882212002E • ios            • iOS 13.4.1
    • macOS         • macOS                     • darwin-x64     • Mac OS X
      10.15.4 19E287
    • Web Server    • web-server                • web-javascript • Flutter Tools
    • Chrome        • chrome                    • web-javascript • Google Chrome
      81.0.4044.138

• No issues found!
``` 

</details>

**Comment by esDotDev** [comment]
There is a chance it is related to this fix: 
https://github.com/flutter/flutter/issues/56071

@ferhatb

**Comment by trevordunn** [comment]
I can confirm the commit this issue starts in is `ba6b3f502d`. And yeah, it’s the commit that #56071 was included in.

**Comment by trevordunn** [comment]
This landed in master with this PR #57150.
